<launch>

  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find rbe500_project)/worlds/scara.world"/>
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <!-- Load the URDF into the ROS Parameter Server -->
  <param name="robot_description"
    command="$(find xacro)/xacro --inorder '$(find rbe500_project)/urdf/scara.xacro'" />

  <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
    args="-urdf -model scara -param robot_description"/>

  <!-- Start forward kinematics node -->
  <!-- node name="forward_kinematics" pkg="rbe500_project" type="forward_kinematics.py" / -->

  <!-- Start inverse kinematics server node -->
  <node name="inverse_kinematics_server" pkg="rbe500_project" type="inverse_kinematics_server.py" output="screen"/>

    <!-- Start position controller nodes -->
  <!-- node name="position_controller" pkg="rbe500_project" type="position_controller.py" output="screen"/ -->
  <node name="position_controller_server" pkg="rbe500_project" type="position_controller_server.py" output="screen"/>


</launch>
